packages/dropdowns/src/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,185 @@
+/**
```

Comment:
Why not create functors instead of declaring the functions inside the component? There is a few of them and they're not small, it makes it harder to understand the logic

Reply:
There is a lot of interaction with the props and items provided via context.  Removing the transforms out of the component was making it personally hard for me to follow the logic.

I'll look into breaking this logic up some more.